pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -125,6 +126,55 @@ func canIgnoreHealthErrStatusInReconcile(cephCluster cephv1.CephCluster, control
 	return ignoreHealthErr
 }
 
+// IsReadyToReconcile determines if a controller is ready to reconcile or not
+func IsReadyToReconcileNoName(c client.Client, clustercontext *clusterd.Context, controllerName string) (cephv1.CephCluster, bool, bool, reconcile.Result) {
+	cephClusterExists := false
+
+	clusterNamespace := os.Getenv(k8sutil.PodNamespaceEnvVar)
```

Comment:
in other places in the code, the following is passed as the namespace: `request.NamespacedName`
is the request namespace the one the cluster is on?

Reply:
The request `request.NamespacedName` is the one from the object that we are reconciling one. When an event is caught and the predicate code return `true` then the request.NamespacedName is one from the object we agreed to reconcile on.

Reply:
ok. so this won't help with the cluster namespace.
anyway, changed the code so that the notification CR will also have the objecstore name and namespace as parameters